llvm/include/llvm/Transforms/IPO/FunctionSpecialization.h

```diff
@@ -217,6 +217,10 @@ class InstCostVisitor : public InstVisitor<InstCostVisitor, Constant *> {
   Cost estimateSwitchInst(SwitchInst &I);
   Cost estimateBranchInst(BranchInst &I);
 
+  bool discoverTransitivelyIncomingValues(
```

Comment:
How about adding comments here so that code readers can understand what is transitive phis and why should we care about them?